### PR TITLE
chore: inline pces proposal

### LINKED ISSUE #13777: Write proposal for elimination of accidental self branches
It's possible for an honest node to create a branch in the hashgraph if it creates a self event, gossips that event, and then crashes prior to that event being written to a PCES file.

Propose a solution that makes it impossible for a node to self-branch in this way.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #13777

### CODE REVIEW

File: platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+| Designers          | Cody, Richard, Jasper               |
+| Functional Impacts | Platform                            |
+| Related Proposals  | N/A                                 |
+| HIPS               | N/A                                 |
+
+---
+
+## Purpose and Context
+
+Nodes in the network create new "self-events" on a regular basis. These self-events carry transactions that were
```
Comment (edward-swirldslabs) [inline platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md:27]: The clear explanation of the mechanism of branching is very appreciated.

File: platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+The `PcesSequencer` will no longer be necessary and can be removed. There will also be no need for keystone events
+within rounds.
+
+#### RoundDurabilityBuffer removal
+
+The `RoundDurabilityBuffer` will no longer be necessary and can be removed.
+
+#### Diagram Changes
+
+In this image, several existing components are crossed out. This represents a component being removed. The data will
```
Comment (edward-swirldslabs) [inline platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md:127]: +1 for the removal off components.

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+still flow along the same pathways, but with no component on that wire.
+
+![](inlinePces.png)](inlinePces.png)
+
+### Core Behaviors
+
+[JEP 352](https://openjdk.org/jeps/352) enhanced the support for `MemoryMappedFile` to support the semantics we need
+for non-volatile memory:
```
Comment (rbair23) [inline platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md:135]: Jasper has been doing more digging, and I'm not sure whether this `ExtendedMapMode.READ_WRITE_SYNC` is right. Or even whether memory mapped files is necessary, or whether a basic FileOutputStream flushing on every self-event is sufficient. I wonder whether the architecture doc should leave off the implementation detail of the PCES box itself, but just say "this box will persist before gossiping self events" or "this box will persist all events before sending them to the hashgraph" and then leave room in the implementation for whatever is found to work.

**Comment by hendrikebbers** [comment]
- 0.5 since the proposal is only understandable for people who worked in the hashgraph team in the past (from my point of view), and the diagrams should have better quality and resolution (inlinePces.png, for example, is missing any description of what is shown in the image).

File: platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md (hunk 1 above)
Comment (kfa-aguda) [reply platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md:27]: indeed! Learning a ton from reading the design docs so far. Yay, design!

File: platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+>    Shared file mappings with this flag provide the guarantee
+>    that while some memory is mapped writable in the address
+>    space of the process, it will be visible in the same file
+>    at the same offset even after the system crashes or is
+>    rebooted.  In conjunction with the use of appropriate CPU
+>    instructions, this provides users of such mappings with a
+>    more efficient way of making data modifications
+>    persistent.
+
+### Performance
```
Comment (kfa-aguda) [inline platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md:183]: does PCES have recovery mechanisms for IO failures? Now that synchronous write to disk will be in the way of consensus (gossip will have to wait for durability of self-events), how will we handle if write doesn't succeed?

Comment (kfa-aguda) [reply platform-sdk/docs/proposals/inlinePces/inlinePcesProposal.md:183]: The only other thing i'd add is could you please enrich the title a bit? Sounds quite different from the description of the proposal.

**Comment by cody-littley** [comment]
Closing this proposal based on Richard's request to first do experimentation with various strategies.

**Comment by cody-littley** [comment]
superseded by https://github.com/hashgraph/hedera-services/pull/14056

**Pull request closed without merging**